### ISSUE TITLE
Mupen64plus crashes when selecting raphnet input plugin

### ISSUE THREAD

**Issue by osaeed18** [loganmc10/m64p#2]
I have a Raphnet N64 to USB adapter - V3 
I am running Mupen64plus build m64p-win64-aff9457 
Mupen64Plus Core' version 2.5.9
When selecting mupen64plus-input-raphnetraw in Mupen input options and loading a game, the mupen64plus application hangs then crashes with the following error.

Windows logs

Log Name:      Application
Source:        Application Error
Date:          2/25/2020 10:28:05 AM
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Description:
Faulting application name: mupen64plus-gui.exe, version: 0.0.0.0, time stamp: 0x5e4d92db
Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000
Exception code: 0xc0000005
Fault offset: 0x0000000000000000
Faulting process id: 0x2d1c
Faulting application start time: 0x01d5ebf01755880f
Faulting application path: D:\LaunchBox\Emulators\Nintendo64\mupen64plus\mupen64plus-gui.exe
Faulting module path: unknown
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Level>2</Level>
    <Task>100</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2020-02-25T15:28:05.589008000Z" />
    <EventRecordID>10497</EventRecordID>
    <Channel>Application</Channel>
    <Security />
  </System>
  <EventData>
    <Data>mupen64plus-gui.exe</Data>
    <Data>0.0.0.0</Data>
    <Data>5e4d92db</Data>
    <Data>unknown</Data>
    <Data>0.0.0.0</Data>
    <Data>00000000</Data>
    <Data>c0000005</Data>
    <Data>0000000000000000</Data>
    <Data>2d1c</Data>
    <Data>01d5ebf01755880f</Data>
    <Data>D:\LaunchBox\Emulators\Nintendo64\mupen64plus\mupen64plus-gui.exe</Data>
    <Data>unknown</Data>
    <Data>975720d3-8797-47b7-81f2-fff9f0b03afa</Data>
    <Data>
    </Data>
    <Data>
    </Data>
  </EventData>
</Event>

Mupen64plus logs

GUI: attached to core library 'Mupen64Plus Core' version 2.5.9
Core: Using full mem base
Core: CRC: 3E5055B6 2E92DA52
Core: Imagetype: .v64 (byteswapped)
Core: Rom size: 12582912 bytes (or 12 Mb or 96 Megabits)
Core: Version: 1446
Core: Manufacturer: Nintendo
Core: Country: USA
GUI: using Video plugin: 'GLideN64' v2.0.0
GUI: using Audio plugin: 'Mupen64Plus SDL2 Audio Plugin' v2.5.0
GUI Error: LoadLibrary('D:/LaunchBox/Emulators/Nintendo64/mupen64plus\mupen64plus-input-raphnetraw.dll') error: The specified module could not be found.

GUI: using RSP plugin: 'Hacktarux/Azimer High-Level Emulation RSP Plugin' v2.5.9
RSP: RSP Fallback disabled !
Core Warning: No input plugin attached.  You won't be able to control the game.
Core: Using video capture backend: dummy
Core: Game controller 0 (Standard controller) has nothing plugged in
Core: Game controller 1 (Standard controller) has nothing plugged in
Core: Game controller 2 (Standard controller) has nothing plugged in
Core: Game controller 3 (Standard controller) has nothing plugged in
Core: Using CIC type X102
Core Warning: OSD not compatible with OpenGL core context. OSD deactivated.
Core: Initializing 4 RDRAM modules for a total of 8 MB
Core: Starting R4300 emulator: Cached Interpreter
RSP Warning: Product Code = 4e4b5445
Core Warning: OSD not compatible with OpenGL core context. OSD deactivated.
Core Warning: OSD not compatible with OpenGL core context. OSD deactivated.

I am using mupen64plus-input-raphnetraw-windows-1.1.0.zip  Version 2019-07-16

OS: Microsoft Windows 10 Pro
OS Version: 10.0.18362 N/A Build 18362

Is this something I should report to raphnet?
Thanks

**Comment by loganmc10** [comment]
No it shouldn't be reported to raphnet, it does work, it's just that the distributed version of mupen64plus-input-raphnetraw.dll doesn't work with the GUI.

I just tested, I can get it to work by compiling it alongside the other GUI components, however I don't have a Raphnet adapter to test. I don't officially support things I can't test (same reason there is no Mac OS X build)

**Comment by loganmc10** [comment]
You can test by using these DLLs:

[Raphnet.zip](https://github.com/loganmc10/m64p/files/4252489/Raphnet.zip)

Just drop them into your mupen64plus folder instead of the dll you were using

**Comment by osaeed18** [comment]
Thank you loganmc10, I have replaced the .dll files and the adapter is correctly recognized and loads the plugin with no sign of errors.

GUI: using Video plugin: 'GLideN64' v2.0.0
GUI: using Audio plugin: 'Mupen64Plus SDL2 Audio Plugin' v2.5.0
GUI: using Input plugin: 'Mupen64Plus raphnetraw Input Plugin' v1.1.0
GUI: using RSP plugin: 'Hacktarux/Azimer High-Level Emulation RSP Plugin' v2.5.9
Core: input plugin did not specify a render callback; there will be no on screen display by the input plugin.
Input: Found USB device 0x289b:0x0061 serial '000001' name 'N64 to USB v3.6'
Input: Adapter supports 1 raw channel(s)
Input: Channel 0: Adapter '000001' raw channel 0
Input: Mupen64Plus raphnetraw Input Plugin version 1.1.0 (compiled Feb 25 2020 15:18:16) initialized.
RSP: RSP Fallback disabled !
Core: Using video capture backend: dummy
Core: Game controller 1 (Standard controller) has nothing plugged in
Core: Game controller 2 (Standard controller) has nothing plugged in
Core: Game controller 3 (Standard controller) has nothing plugged in
Core: Using CIC type X102
